pkg/query-service/rules/manager.go

```diff
@@ -233,7 +253,24 @@ func (m *Manager) EditRule(ruleStr string, id string) error {
 	}
 
 	if !m.opts.DisableRules {
-		return m.syncRuleStateWithTask(taskName, parsedRule)
+		err = m.syncRuleStateWithTask(taskName, parsedRule)
+		if err != nil {
+			return err
+		}
+	}
+
+	// update feature usage if the current rule is not a trace or log query builder
+	if !checkIfTraceOrLogQB(&currentRule.PostableRule) {
+		err = m.updateFeatureUsage(parsedRule, 1)
+		if err != nil {
+			zap.S().Errorf("error updating feature usage: %v", err)
+		}
```

Comment:
`currentRule` here is the existing record from DB. What does this _update feature usage if the current rule is not a trace or log query builder_ mean? Why do we need to update feature usage in that case?

Reply:
If currentRule is not trace/logs then we call updateFeatureUsage() which updates feature if new rule is trace/logs.

If currentRule is trace/logs and new rule is not trace/logs then we call updateFeatureUsage() which updates feature if current rule is trace/logs.